pkg/test/framework/environment/environment.go

```diff
@@ -0,0 +1,207 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package environment
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/gogo/protobuf/proto"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/framework/settings"
+)
+
+const (
+	httpOK = "200"
+)
+
+type (
+
+	// Environment is the public interface for interacting with the testing environment. It provides a
+	// convenient facade that allows quickly accessing individual components.
+	Environment interface {
```

Comment:
Should we have a more generic `GetComponent` method as well? This would allow custom components without having to change the interface

Reply:
That still requires a bunch of leg-work for registration and keeping it local. The only thing you'd end up saving is updating the environment facade which, at this point, is trivial.